llvm/include/llvm/SYCLLowerIR/GlobalOffset.h

```diff
@@ -5,21 +5,117 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
-//
-// This pass operates on SYCL kernels being compiled to CUDA. It looks for uses
-// of the `llvm.nvvm.implicit.offset` intrinsic and replaces it with a offset
-// parameter which will be threaded through from the kernel entry point.
-//
-//===----------------------------------------------------------------------===//
 
 #ifndef LLVM_SYCL_GLOBALOFFSET_H
 #define LLVM_SYCL_GLOBALOFFSET_H
 
-#include "llvm/Pass.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/SYCLLowerIR/TargetHelpers.h"
 
 namespace llvm {
 
-ModulePass *createGlobalOffsetPass();
+class ModulePass;
+class PassRegistry;
+
+/// This pass operates on SYCL kernels. It looks for uses of the
+/// `llvm.{amdgcn|nvvm}.implicit.offset` intrinsic and replaces it with an
+/// offset parameter which will be threaded through from the kernel entry
+/// point.
+class GlobalOffsetPass : public PassInfoMixin<GlobalOffsetPass> {
+private:
+  using KernelPayload = TargetHelpers::KernelPayload;
+  using ArchType = TargetHelpers::ArchType;
+
+public:
+  explicit GlobalOffsetPass() {}
+
+  PreservedAnalyses run(Module &M, ModuleAnalysisManager &);
+  static StringRef getPassName() { return "Add implicit SYCL global offset"; }
+
+private:
+  /// After the execution of this function, the module to which the kernel
+  /// `Func` belongs contains a clone of the original kernel with the signature
```

Comment:
Can you please specify if the original kernel is deleted or not? Thanks

Reply:
It is not, will expand the comment.